cdap-watchdog/src/main/java/co/cask/cdap/logging/save/LogMetricsPlugin.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.logging.save;
+
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.common.logging.LoggingContext;
+import co.cask.cdap.common.metrics.MetricsCollectionService;
+import co.cask.cdap.common.metrics.MetricsCollector;
+import co.cask.cdap.logging.appender.kafka.KafkaTopic;
+import co.cask.cdap.logging.context.LoggingContextHelper;
+import co.cask.cdap.logging.kafka.KafkaLogEvent;
+import co.cask.tephra.TransactionExecutorFactory;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Throwables;
+import com.google.common.collect.Maps;
+import com.google.common.util.concurrent.ListeningScheduledExecutorService;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.google.inject.Inject;
+import org.apache.twill.common.Threads;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Plugin to process Log and generate metrics on Log level.
+ */
+public class LogMetricsPlugin extends AbstractKafkaLogProcessor {
+
+  private final MetricsCollectionService metricsCollectionService;
+
+  public static final String SYSTEM_METRIC_PREFIX = "services.log";
+  public static final String APP_METRIC_PREFIX = "app.log";
+  private final CheckpointManager checkpointManager;
+  private static final int ROW_KEY_PREFIX = 101;
```

Comment:
It is a good idea to put all static final in the beginning. And LOG should be the first one.